apbs/ainput/apbs_legacy_input.py

```diff
@@ -0,0 +1,1203 @@
+#   filenames/pathnames.
+
+
+class ApbsLegacyInput:
+    """Class for reading in legacy APBS input files."""
+
+    def __init__(self):
+        """Setup parsing tokens and grammar for the APBS input file format."""
+
+        # Tokens used by multiple parsers
+        self.COMMENT = "#"
+        # TODO: The DEBUG flag should be set/accessed from a higher
+        #       level debugging object that is set from a config
+        #       file or command line option.
+        self.DEBUG = False
+        self.END_VAL = CLiteral("END")
+
+        # The FINAL_OUTPUT is a dictionary produced after parsing a file
+        # using the grammar and processing rules.
+        self.FINAL_OUTPUT = {}
+
+        # The highest level grammar to parse the READ, ELEC, APOLAR, and
+        # PRINT sections until the QUIT keyword is found.
+        self.grammar = OneOrMore(self.readParser()) + OneOrMore(
+            self.apolarParser() | self.elecParser()
+        ) + ZeroOrMore(self.printParser()) + Suppress(CLiteral("QUIT")) | (
+            # The following is one way to "narrow down" errors in an input
+            # file. It was adopted from:
+            # https://web.archive.org/web/20160821175151/http://pyparsing.wikispaces.com/share/view/30875955
+            # The idea is that the parsers above will succeed or the
+            # following will succeed and set the ParseSyntaxException
+            # stating that the token in the file is "unknown" and will
+            # show the line and column where the parser stopped. It does
+            # not always point out subtle errors like when a key is used
+            # in a section that does not support it. For example, using
+            # a keyword from pbam-auto in an ELEC section that specifies
+            # using tabi-auto type.
+            empty
+            - ~Word(printables).setName("<unknown>")
+        )
+
+    @staticmethod
+    def get_integer_grammar():
+        """Convenience function that is used by other classes."""
+        return Combine(Optional("-") + Word(nums))  # .setDebug()
+
+    @staticmethod
+    def get_identifier_grammar():
+        """Convenience function that is used by other classes."""
+        return (
+            Word(alphanums, alphanums + r"_" + r"-")
+            | ApbsLegacyInput.get_integer_grammar()
+        )  # .setDebug()
+
+    @staticmethod
+    def get_number_grammar():
+        """Convenience function that is used by other classes."""
+        return (
+            pyparsing_common.real | ApbsLegacyInput.get_integer_grammar()
+        )  # .setDebug()
+
+    @staticmethod
+    def get_path_grammar():
+        """Convenience function that is used by other classes."""
+        return Word(printables)  # .setDebug()
+
+    def debug(self, message: str):
+        """Convenience function to track down grammar/parsing problems"""
+        if self.DEBUG:
+            print(message)
+
+    def formatReadSection(self, results: ParseResults, groups: list) -> dict:
```

Comment:
This doesn't follow PEP8 style conventions:  https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

Reply:
I will make the changes to all the class and function names. I thought flake8 would catch this stuff and many of the functions and classnames in pyparsing don't follow that convention.

Reply:
Thanks -- I've had to use a combination of pylint and flake8 to catch everything.